server/src/test/java/org/elasticsearch/index/query/functionscore/FunctionScoreQueryBuilderTests.java

```diff
@@ -837,21 +837,22 @@ public void testCacheability() throws IOException {
         assertEquals("query should " + (isCacheable ? "" : "not") + " be cacheable: " + queryBuilder.toString(), isCacheable,
                 context.isCacheable());
 
-        // check the two non-cacheable cases explicitly
         ScoreFunctionBuilder<?> scriptScoreFunction = new ScriptScoreFunctionBuilder(
                 new Script(ScriptType.INLINE, MockScriptEngine.NAME, "1", Collections.emptyMap()));
-        RandomScoreFunctionBuilder randomScoreFunctionBuilder = new RandomScoreFunctionBuilderWithFixedSeed();
-
-        for (ScoreFunctionBuilder<?> scoreFunction : List.of(scriptScoreFunction, randomScoreFunctionBuilder)) {
-            FilterFunctionBuilder[] functions = new FilterFunctionBuilder[] {
-                    new FilterFunctionBuilder(RandomQueryBuilder.createQuery(random()), scoreFunction) };
-            queryBuilder = new FunctionScoreQueryBuilder(functions);
+        queryBuilder = new FunctionScoreQueryBuilder(new FilterFunctionBuilder[] {
+            new FilterFunctionBuilder(RandomQueryBuilder.createQuery(random()), scriptScoreFunction) });
+        context = createShardContext();
+        rewriteQuery = rewriteQuery(queryBuilder, new QueryShardContext(context));
+        assertNotNull(rewriteQuery.toQuery(context));
+        assertTrue("script should query should be cacheable: " + queryBuilder.toString(), context.isCacheable());
 
-            context = createShardContext();
-            rewriteQuery = rewriteQuery(queryBuilder, new QueryShardContext(context));
-            assertNotNull(rewriteQuery.toQuery(context));
-            assertFalse("query should not be cacheable: " + queryBuilder.toString(), context.isCacheable());
-        }
+        RandomScoreFunctionBuilder randomScoreFunctionBuilder = new RandomScoreFunctionBuilderWithFixedSeed();
+        queryBuilder = new FunctionScoreQueryBuilder(new FilterFunctionBuilder[] {
+            new FilterFunctionBuilder(RandomQueryBuilder.createQuery(random()), randomScoreFunctionBuilder) });
+        context = createShardContext();
+        rewriteQuery = rewriteQuery(queryBuilder, new QueryShardContext(context));
+        assertNotNull(rewriteQuery.toQuery(context));
+        assertFalse("random score should query should not be cacheable: " + queryBuilder.toString(), context.isCacheable());
```

Comment:
"random score should query should not be cacheable: "  -> "function random query should not be cacheable: "

Reply:
We also need to remove `ScriptScoreFunctionBuilder` as a condition for `false` in the `isCacheable` function (line 861-862), as for `ScriptScoreFunctionBuilder` case we want to return `true`.

Reply:
Updated message and removed `ScriptScoreFunctionBuilder`.